File: R/chi_square_mendelian_test.R
observed <- c(dominant = 315, recessive = 105)
expected <- c(dominant = 420 * 3/4, recessive = 420 * 1/4)

chi_square <- sum((observed - expected)^2 / expected)

cat("Observed counts\n")
print(observed)

cat("Expected counts\n")
print(expected)

cat("Chi-square =", round(chi_square, 4), "\n")

File: python/chi_square_mendelian_test.py
observed = {"dominant": 315, "recessive": 105}
total = sum(observed.values())

expected = {
    "dominant": total * 3/4,
    "recessive": total * 1/4
}

chi_square = sum(
    (observed[k] - expected[k])**2 / expected[k]
    for k in observed
)

print("Observed counts:", observed)
print("Expected counts:", expected)
print("Chi-square =", round(chi_square, 4))
